peps/pep-0750.rst

```diff
@@ -53,837 +33,1089 @@ deferred pending further experience with f-strings. Work on this PEP was
-interpolated expressions had access to regular scoping rules. However, f-strings have
-two main limitations - expressions are eagerly evaluated, and interpolated values
-cannot be intercepted. The former means that f-strings cannot be re-used like templates,
-and the latter means that how values are interpolated cannot be customized.
-
-Templating in Python is currently achieved using packages like Jinja2 which bring their
-own templating languages for generating dynamic content. In addition to being one more
-thing to learn, these languages are not nearly as expressive as Python itself. This
-means that business logic, which cannot be expressed in the templating language, must be
-written in Python instead, spreading the logic across different languages and files.
-
-Likewise, the inability to intercept interpolated values means that they cannot be
-sanitized or otherwise transformed before being integrated into the final string. Here,
-the convenience of f-strings could be considered a liability. For example, a user
-executing a query with `sqlite3 <https://docs.python.org/3/library/sqlite3.html>`__
-may be tempted to use an f-string to embed values into their SQL expression instead of
-using the ``?`` placeholder and passing the values as a tuple to avoid an
-`SQL injection attack <https://en.wikipedia.org/wiki/SQL_injection>`__.
-
-Tag strings address both these problems by extending the f-string syntax to provide
-developers access to the string and its interpolated values before they are combined. In
-doing so, tag strings may be interpreted in many different ways, opening up the
-possibility for DSLs and other custom string processing.
-
-Proposal
-========
-
-This PEP proposes customizable prefixes for f-strings. These f-strings then
-become a "tag string": an f-string with a "tag function." The tag function is
-a callable which is given a sequence of arguments for the parsed tokens in
-the string.
+Python f-strings are easy to use and very popular. Over time, however, developers
+have encountered limitations that make them unsuitable for certain use cases. 
+In particular, f-strings provide no way to intercept and transform interpolated 
+values before they are combined into a final string.
 
-Here's a very simple example. Imagine we want a certain kind of string with
-some custom business policies: uppercase the value and add an exclamation point.
-
-Let's start with a tag string which simply returns a static greeting:
-
-.. code-block:: python
+As a result, incautious use of f-strings can lead to security vulnerabilities. 
+For example, a user executing a SQL query with `sqlite3 <https://docs.python.org/3/library/sqlite3.html>`__
+may be tempted to use an f-string to embed values into their SQL expression 
+instead of using the ``?`` placeholder and passing the values as a tuple to 
+avoid a `SQL injection attack <https://en.wikipedia.org/wiki/SQL_injection>`__.
```

Comment:
I don't think mentioning `?` has much value, the point is more that f-strings can be bad. 

```suggestion
As a result, incautious use of f-strings can lead to security vulnerabilities. 
For example, a user executing an SQL query with :mod:`python:sqlite3`
may be tempted to use an f-string to embed values into their SQL expression,
which can be vulnerable to an `SQL injection attack <https://en.wikipedia.org/wiki/SQL_injection>`__.
```

Reply:
Agree, this is strictly better.